libs/shared/ui/src/lib/components/link/link.tsx

```diff
@@ -1,58 +1,92 @@
-  iconLeftClassName?: string
-}
+const linkVariants = cva(
+  ['cursor-pointer', 'transition', 'duration-100', 'font-medium', 'inline-flex', 'flex-center', 'gap-1'],
+  {
+    variants: {
+      color: {
+        brand: ['text-brand-500', 'hover:text-brand-600'],
+        red: ['text-red-500', 'hover:text-red-600'],
+        sky: ['text-sky-500', 'hover:text-sky-600'],
+      },
+      size: {
+        xs: ['text-xs'],
+        ssm: ['text-ssm'],
+        sm: ['text-sm'],
+      },
+    },
+    defaultVariants: {
+      color: 'sky',
+      size: 'sm',
+    },
+  }
+)
 
-export function Link(props: LinkProps) {
-  const {
-    link,
-    linkLabel,
-    external = false,
-    className = '',
-    size = 'text-sm',
-    iconLeft,
-    iconRight,
-    iconLeftClassName = 'text-xs leading-5',
-    iconRightClassName = 'ml-0.5 text-xs leading-5 ',
-  } = props
+const iconVariants = cva([], {
+  variants: {
+    size: {
+      xs: ['text-2xs', 'leading-4'],
+      ssm: ['text-xs', 'leading-5'],
+      sm: ['text-xs', 'leading-5'],
+    },
+  },
+  defaultVariants: {
+    size: 'sm',
+  },
+})
 
-  const currentClassName = `${className} ${size} text-sky-500 inline-flex flex-center gap-1 hover:underline`
+export interface ExternalLinkProps
+  extends Omit<ComponentPropsWithoutRef<'a'>, 'color'>,
+    VariantProps<typeof linkVariants> {
+  className?: string
+}
 
-  const content = () => (
-    <>
-      {iconLeft && <Icon name={iconLeft} className={iconLeftClassName} />}
-      {linkLabel}
-      {iconRight && <Icon name={iconRight} className={iconRightClassName} />}
-    </>
+export const ExternalLink = forwardRef<HTMLAnchorElement, ExternalLinkProps>(function Link(
+  { children, color, size, className, ...props },
+  forwardedRef
+) {
+  return (
+    <a
+      ref={forwardedRef}
+      className={twMerge(linkVariants({ color, size }), className)}
+      target="_blank"
+      rel="noopener noreferrer"
+      {...props}
+    >
+      {children}
+      <Icon name={IconAwesomeEnum.ARROW_UP_RIGHT_FROM_SQUARE} className={iconVariants({ size })} />
+    </a>
   )
+})
 
-  if (external) {
-    return (
-      <a className={currentClassName} href={link} target="_blank" rel="noreferrer">
-        {content()}
-      </a>
-    )
-  } else {
-    return (
-      <NavLink className={currentClassName} to={link}>
-        {content()}
-      </NavLink>
-    )
-  }
+export interface LinkProps extends Omit<NavLinkProps, 'color'>, VariantProps<typeof linkVariants> {
+  className?: string
+  icon?: IconAwesomeEnum | string
 }
 
-export default Link
+export const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link(
+  { children, color, size, className, icon, ...props },
+  forwardedRef
+) {
+  return (
+    <NavLink ref={forwardedRef} className={twMerge(linkVariants({ color, size }), className)} {...props}>
+      {/* Fragment is necessary */}
+      <>
+        {children}
+        {icon && <Icon name={icon} className={iconVariants({ size })} />}
```

Comment:
Not sure we want to couple those `Icon` and `Link`.
I prefer:
```
<Link to="/" size="xs">
  Foobar
  <Icon size="xs">
</Link>
```
You're not stuck with icon positioning

Reply:
I understand your point but for it, I need to refactor the Icon component as well
Maybe done later, but I'm in favor of leaving it like that. And it's not easy like in your example, because the font size of the text is not the same as Icon, the icon is one size smaller